homeassistant/components/home_connect/binary_sensor.py

```diff
@@ -86,9 +94,16 @@ async def async_update(self) -> None:
                 "Unexpected value for HomeConnect %s state: %s", self._type, state
             )
             self._state = None
-        _LOGGER.debug("Updated, new state: %s", self._state)
+        _LOGGER.debug(
+            "Updated value for HomeConnect %s, new state: %s", self._type, self._state
+        )
+
+    @property
+    def icon(self) -> str | None:
+        """Return the icon."""
+        return self._icon
 
     @property
-    def device_class(self):
+    def device_class(self) -> BinarySensorDeviceClass | None:
```

Comment:
This is refactoring that is not related to this PR, please remove all unrelated changes.